swayfmt/src/formatter/mod.rs

```diff
@@ -15,11 +15,13 @@ pub(crate) mod shape;
 
 #[derive(Debug, Default, Clone)]
 pub struct Formatter {
+    pub src: UnformattedCode,
```

Comment:
Hmmm we probably don't want this, as for items that have multiple children, the first child will overwrite this field before the second is traversed, and so on.

What's the aim behind storing this? Don't we already receive it as an argument?

Reply:
The formatter receives it as an input into `format()`.

The motivation right now is just to use it to look for newlines between comments [here](https://github.com/FuelLabs/sway/pull/4005/files#diff-f6bea861eb458d4938d8c83563d84ccacd610d1af47a750e001ab26fdf9c79efR11-R15). I think having the original src to look at during a formatting pass would be useful in situations like this

Reply:
If we want to store the top-level unformatted code in the `Formatter`, then this should be done as a part of the `Formatter`'s construction and the field should not be `pub` to ensure any implementions of `Format` do not overwrite this field during traversal. We should probably also add a comment explaining this is the full, unformatted code.